torch_xla/experimental/xla_sharding.py

```diff
@@ -1,14 +1,74 @@
+from collections import OrderedDict
 import torch
 import torch_xla
 import torch_xla.core.xla_model as xm
 from torch_xla.experimental.xla_sharded_tensor import XLAShardedTensor
+from torch_xla.experimental.pjrt import requires_pjrt
 
 import numpy as np
-from typing import Tuple, Union
+from typing import Tuple, Union, List
 
 
-def mark_sharding(t: Union[torch.Tensor,
-                           XLAShardedTensor], mesh_shape: Tuple[int],
+class Mesh:
+  """Describe the logical XLA device topology mesh and the underlying resources.
+
+  Args:
+    device_ids (Union[np.ndarray, List]): A raveled list of devices (IDs) in a custom order. The list is reshaped
+        to an `mesh_shape` array, filling the elements using C-like index order. For example,
```

Comment:
where is the example lol?

Reply:
oh ok it is below, you might want to change the wording here.

Reply:
Done